hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerManager.java

```diff
@@ -107,7 +108,21 @@ public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerStatus(
     List<DatanodeDetails> filterDns = null;
     if (hosts.isPresent() && !hosts.get().isEmpty()) {
       filterDns = NodeUtils.mapHostnamesToDatanodes(nodeManager, hosts.get(),
-          useHostnames);
+          useHostnames).stream()
+          .filter(dn -> {
+            try {
+              NodeStatus nodeStatus = nodeManager.getNodeStatus(dn);
+              if (nodeStatus != NodeStatus.inServiceHealthy()) {
+                LOG.warn("Datanode {} is not in optimal state for disk balancing." +
+                    " Operational state: {}", dn.getHostName(), nodeStatus.getOperationalState());
```

Comment:
```suggestion
                    " Operational state: {}", dn.getHostName(), nodeStatus);
```
How about print the whole status? as we filter base on both health and operation state.

Reply:
No issue. It's completly alright.

Reply:
> Also, do we have test for this command? If not, then I think it's ok to merge this, otherwise we can test the prodution code update.

`TestDiskBalancerManager.testDatanodeDiskBalancerStatus()` this is to test wether status is displayed for all DN.
Probably we can update this unit test to also check when a specified host status is queried which is not in service then nothing should be displayed.
I think this will be good.

Reply:
I think it's great. Thanks for the info!

Reply:
yes you are correct.

Reply:
Then I think we can only test if those not health nodes are filtered.